src/ControlSystem/FutureMeasurements.cpp

```diff
@@ -0,0 +1,65 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/FutureMeasurements.hpp"
+
+#include <limits>
+#include <pup_stl.h>
+
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace control_system {
+FutureMeasurements::FutureMeasurements(const size_t measurements_per_update,
+                                       const double first_measurement_time)
+    : measurements_{std::numeric_limits<double>::signaling_NaN(),
+                    first_measurement_time},
+      measurements_until_update_(measurements_per_update - 1),
```

Comment:
```cpp
ASSERT(measurements_per_update > 0, ...)
```
? Obviously other things should break if this was actually false, but would be good as a safety check.